tgui/packages/tgui/interfaces/common/WirePanel/index.tsx

```diff
@@ -0,0 +1,190 @@
+      { (control_lights & WirePanelControls.WIRE_CONTROL_SILICON) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_SILICON]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_SILICON && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_ACCESS) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_ACCESS]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_ACCESS && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_SAFETY) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_SAFETY]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_SAFETY && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_LIMITER) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_LIMITER]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_LIMITER && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_TRIGGER) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_TRIGGER]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_TRIGGER && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_RECEIVE) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_RECEIVE]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_RECEIVE && "On" || "Off"} </Box>}
+      { (control_lights & WirePanelControls.WIRE_CONTROL_TRANSMIT) && <Box className="indicator_label">{WirePanelControlLabels[WirePanelControls.WIRE_CONTROL_TRANSMIT]}: {hacked_controls & WirePanelControls.WIRE_CONTROL_TRANSMIT && "On" || "Off"} </Box>}
+    </Box>
+  );
+};
+
+// remote silicon wire checks
+const RemoteAccessButton = (props, context) => {
+  const {
+    disabled,
+    ...rest
+  } = props;
+  const { data } = useBackend<WirePanelData>(context);
+  const { is_silicon_user, hacked_controls, is_accessing_remotely, cover_status } = data.wire_panel;
+  let shouldDisable = false;
+  if (is_silicon_user) {
+    if ((hacked_controls & WirePanelControls.WIRE_CONTROL_SILICON)) {
+      if (is_accessing_remotely) {
+        shouldDisable = true;
+      } else {
+        shouldDisable = cover_status !== WirePanelCoverStatus.WPANEL_COVER_OPEN;
+      }
+    }
+  } else {
+    shouldDisable = cover_status !== WirePanelCoverStatus.WPANEL_COVER_OPEN;
+  }
+  return (
+    <Button
+      className="WirePanel-wires-generic-button"
+      disabled={shouldDisable}
+      {...rest}
+    />
+  );
+};
+
+// visually block the entire TGUI window based on silicon control
+export const RemoteAccessBlocker = (is_accessing_remotely: BooleanLike, hacked_controls: number) => {
```

Comment:
```suggestion
interface RemoteAccessBlockerProps {
  is_accessing_remotely: BooleanLike;
  hacked_controls: number;
}

export const RemoteAccessBlocker = ({ is_accessing_remotely, hacked_controls }: RemoteAccessBlockerProps) => {
```
As this is a component, take props rather than arguments.